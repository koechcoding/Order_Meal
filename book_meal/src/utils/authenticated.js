export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}