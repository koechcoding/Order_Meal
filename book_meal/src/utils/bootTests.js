import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
