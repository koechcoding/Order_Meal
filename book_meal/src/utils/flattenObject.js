export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}