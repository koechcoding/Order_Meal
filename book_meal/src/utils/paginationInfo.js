export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}