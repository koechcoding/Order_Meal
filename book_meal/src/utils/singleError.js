const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;