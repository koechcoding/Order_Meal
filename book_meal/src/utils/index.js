import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}