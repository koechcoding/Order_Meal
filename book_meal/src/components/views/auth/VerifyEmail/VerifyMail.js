import React from 'react';
import axios from 'src/axios';
import { Alert } from 'reactstrap';
import { Redirect, Link } from 'react-router-dom';
import AuthPage from 'src/components/common/AuthPage';
import { Role } from 'src/constants';
import { authenticated, singleError } from 'src/utils';

class VerifyMail extends React.Component {

    state = {}

    componentWillMount() {
        const { match } = this.props;
        const token = match ? match.params.token : null;
        if (token) {
            this.setState({
                ...this.state,
                token,
            });
            axios.post('auth/verify-email', { token }).then(({ data }) => {
                this.setState({
                    ...this.state,
                    success: true,
                })
            }).catch(({ response }) => {
                this.setState({
                    ...this.state,
                    error: response
                });
            })
        } else {
            this.setState({
                ...this.state,
                error: {
                    message: 'Email verification token is required!'
                }
            });
        }
    }

    render() {
        const { error, success } = this.state;
        const user = authenticated();
        if (user) {
            switch(user.role) {
                case Role.SUPER_ADMIN:
                case Role.ADMIN:
                    return <Redirect to="/admin/meals" />;
                case Role.USER:
                    return <Redirect to="/user/menu" />;
                default:
                    break;
            }
        }

        let loading = false;
        if (! error && ! success) 
            loading = true;

        return (
            <AuthPage styles={{ minHeight: '500px', height: '100vh' }} loading={loading}>
                <div 
                    style={{ minHeight: '280px' }}
                    className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 text-center card">

                    <h5 className="text-center mb-3 mt-4">Email Verification.</h5>
                    {error && 
                            <div>
                                <Alert color="danger"> {singleError(error)} </Alert>
                                <p className="p-2">
                                    An error occured while verifying your email account,
                                    please try again.
                                </p>
                            </div>
                    }
                    {success && 
                        <div>
                            <Alert color="success">
                                Verification successful!
                            </Alert>
                            <p className="p-2">
                                You have successfully verified your email address.
                                Please proceed to login.
                            </p>
                            <Link to='/login' className="btn btn-primary w-50 mx-auto">Login</Link>
                        </div>
                    }

                    {!error && !success &&
                        <p className="p-2">
                            Verifying your email address...
                        </p>
                    }
                </div>
            </AuthPage>
        );
    }
}

export default VerifyMail;