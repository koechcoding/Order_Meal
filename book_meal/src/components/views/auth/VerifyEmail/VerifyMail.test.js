import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; // order matters!
import VerifyMail from './VerifyMail';

describe('<VerifyMail />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<VerifyMail/>);
        expect(wrapper).toMatchSnapshot();
    });
});