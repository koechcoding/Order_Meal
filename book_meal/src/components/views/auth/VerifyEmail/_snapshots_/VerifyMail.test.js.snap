// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifyMail /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerifyMail />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 text-center card"
        style={
                Object {
                        "minHeight": "280px",
                      }
        }
>
        <h5
                className="text-center mb-3 mt-4"
        >
                Email Verification.
        </h5>
        <div>
                <Alert
                        closeAriaLabel="Close"
                        color="danger"
                        fade={true}
                        isOpen={true}
                        tag="div"
                        transition={
                                Object {
                                        "appear": true,
                                        "baseClass": "fade",
                                        "baseClassActive": "show",
                                        "enter": true,
                                        "exit": true,
                                        "in": true,
                                        "mountOnEnter": false,
                                        "onEnter": [Function],
                                        "onEntered": [Function],
                                        "onEntering": [Function],
                                        "onExit": [Function],
                                        "onExited": [Function],
                                        "onExiting": [Function],
                                        "tag": "div",
                                        "timeout": 150,
                                        "unmountOnExit": true,
                                      }
                        }
                >
                         
                        Unknown error. Please try again.
                         
                </Alert>
                <p
                        className="p-2"
                >
                        An error occured while verifying your email account, please try again.
                </p>
        </div>
</div>,
      "loading": false,
      "styles": Object {
        "height": "100vh",
        "minHeight": "500px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="text-center mb-3 mt-4"
>
            Email Verification.
</h5>,
          <div>
            <Alert
                        closeAriaLabel="Close"
                        color="danger"
                        fade={true}
                        isOpen={true}
                        tag="div"
                        transition={
                                    Object {
                                                "appear": true,
                                                "baseClass": "fade",
                                                "baseClassActive": "show",
                                                "enter": true,
                                                "exit": true,
                                                "in": true,
                                                "mountOnEnter": false,
                                                "onEnter": [Function],
                                                "onEntered": [Function],
                                                "onEntering": [Function],
                                                "onExit": [Function],
                                                "onExited": [Function],
                                                "onExiting": [Function],
                                                "tag": "div",
                                                "timeout": 150,
                                                "unmountOnExit": true,
                                              }
                        }
            >
                         
                        Unknown error. Please try again.
                         
            </Alert>
            <p
                        className="p-2"
            >
                        An error occured while verifying your email account, please try again.
            </p>
</div>,
          undefined,
          false,
        ],
        "className": "col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 text-center card",
        "style": Object {
          "minHeight": "280px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email Verification.",
            "className": "text-center mb-3 mt-4",
          },
          "ref": null,
          "rendered": "Email Verification.",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Alert
                closeAriaLabel="Close"
                color="danger"
                fade={true}
                isOpen={true}
                tag="div"
                transition={
                                Object {
                                                "appear": true,
                                                "baseClass": "fade",
                                                "baseClassActive": "show",
                                                "enter": true,
                                                "exit": true,
                                                "in": true,
                                                "mountOnEnter": false,
                                                "onEnter": [Function],
                                                "onEntered": [Function],
                                                "onEntering": [Function],
                                                "onExit": [Function],
                                                "onExited": [Function],
                                                "onExiting": [Function],
                                                "tag": "div",
                                                "timeout": 150,
                                                "unmountOnExit": true,
                                              }
                }
>
                 
                Unknown error. Please try again.
                 
</Alert>,
              <p
                className="p-2"
>
                An error occured while verifying your email account, please try again.
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  "Unknown error. Please try again.",
                  " ",
                ],
                "closeAriaLabel": "Close",
                "color": "danger",
                "fade": true,
                "isOpen": true,
                "tag": "div",
                "transition": Object {
                  "appear": true,
                  "baseClass": "fade",
                  "baseClassActive": "show",
                  "enter": true,
                  "exit": true,
                  "in": true,
                  "mountOnEnter": false,
                  "onEnter": [Function],
                  "onEntered": [Function],
                  "onEntering": [Function],
                  "onExit": [Function],
                  "onExited": [Function],
                  "onExiting": [Function],
                  "tag": "div",
                  "timeout": 150,
                  "unmountOnExit": true,
                },
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Unknown error. Please try again.",
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "An error occured while verifying your email account, please try again.",
                "className": "p-2",
              },
              "ref": null,
              "rendered": "An error occured while verifying your email account, please try again.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        undefined,
        false,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 text-center card"
          style={
                    Object {
                              "minHeight": "280px",
                            }
          }
>
          <h5
                    className="text-center mb-3 mt-4"
          >
                    Email Verification.
          </h5>
          <div>
                    <Alert
                              closeAriaLabel="Close"
                              color="danger"
                              fade={true}
                              isOpen={true}
                              tag="div"
                              transition={
                                        Object {
                                                  "appear": true,
                                                  "baseClass": "fade",
                                                  "baseClassActive": "show",
                                                  "enter": true,
                                                  "exit": true,
                                                  "in": true,
                                                  "mountOnEnter": false,
                                                  "onEnter": [Function],
                                                  "onEntered": [Function],
                                                  "onEntering": [Function],
                                                  "onExit": [Function],
                                                  "onExited": [Function],
                                                  "onExiting": [Function],
                                                  "tag": "div",
                                                  "timeout": 150,
                                                  "unmountOnExit": true,
                                                }
                              }
                    >
                               
                              Unknown error. Please try again.
                               
                    </Alert>
                    <p
                              className="p-2"
                    >
                              An error occured while verifying your email account, please try again.
                    </p>
          </div>
</div>,
        "loading": false,
        "styles": Object {
          "height": "100vh",
          "minHeight": "500px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="text-center mb-3 mt-4"
>
              Email Verification.
</h5>,
            <div>
              <Alert
                            closeAriaLabel="Close"
                            color="danger"
                            fade={true}
                            isOpen={true}
                            tag="div"
                            transition={
                                          Object {
                                                        "appear": true,
                                                        "baseClass": "fade",
                                                        "baseClassActive": "show",
                                                        "enter": true,
                                                        "exit": true,
                                                        "in": true,
                                                        "mountOnEnter": false,
                                                        "onEnter": [Function],
                                                        "onEntered": [Function],
                                                        "onEntering": [Function],
                                                        "onExit": [Function],
                                                        "onExited": [Function],
                                                        "onExiting": [Function],
                                                        "tag": "div",
                                                        "timeout": 150,
                                                        "unmountOnExit": true,
                                                      }
                            }
              >
                             
                            Unknown error. Please try again.
                             
              </Alert>
              <p
                            className="p-2"
              >
                            An error occured while verifying your email account, please try again.
              </p>
</div>,
            undefined,
            false,
          ],
          "className": "col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 text-center card",
          "style": Object {
            "minHeight": "280px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email Verification.",
              "className": "text-center mb-3 mt-4",
            },
            "ref": null,
            "rendered": "Email Verification.",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Alert
                  closeAriaLabel="Close"
                  color="danger"
                  fade={true}
                  isOpen={true}
                  tag="div"
                  transition={
                                    Object {
                                                      "appear": true,
                                                      "baseClass": "fade",
                                                      "baseClassActive": "show",
                                                      "enter": true,
                                                      "exit": true,
                                                      "in": true,
                                                      "mountOnEnter": false,
                                                      "onEnter": [Function],
                                                      "onEntered": [Function],
                                                      "onEntering": [Function],
                                                      "onExit": [Function],
                                                      "onExited": [Function],
                                                      "onExiting": [Function],
                                                      "tag": "div",
                                                      "timeout": 150,
                                                      "unmountOnExit": true,
                                                    }
                  }
>
                   
                  Unknown error. Please try again.
                   
</Alert>,
                <p
                  className="p-2"
>
                  An error occured while verifying your email account, please try again.
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    " ",
                    "Unknown error. Please try again.",
                    " ",
                  ],
                  "closeAriaLabel": "Close",
                  "color": "danger",
                  "fade": true,
                  "isOpen": true,
                  "tag": "div",
                  "transition": Object {
                    "appear": true,
                    "baseClass": "fade",
                    "baseClassActive": "show",
                    "enter": true,
                    "exit": true,
                    "in": true,
                    "mountOnEnter": false,
                    "onEnter": [Function],
                    "onEntered": [Function],
                    "onEntering": [Function],
                    "onExit": [Function],
                    "onExited": [Function],
                    "onExiting": [Function],
                    "tag": "div",
                    "timeout": 150,
                    "unmountOnExit": true,
                  },
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Unknown error. Please try again.",
                  " ",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "An error occured while verifying your email account, please try again.",
                  "className": "p-2",
                },
                "ref": null,
                "rendered": "An error occured while verifying your email account, please try again.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          undefined,
          false,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;